src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -32,12 +34,21 @@ class ExpressionLanguage
     protected $functions = array();
 
     /**
-     * @param ParserCacheInterface                  $cache
+     * @param CacheItemPoolInterface                $cache
      * @param ExpressionFunctionProviderInterface[] $providers
      */
-    public function __construct(ParserCacheInterface $cache = null, array $providers = array())
+    public function __construct($cache = null, array $providers = array())
     {
-        $this->cache = $cache ?: new ArrayParserCache();
+        if (null !== $cache) {
+            if ($cache instanceof ParserCacheInterface) {
+                @trigger_error(sprintf('Passing an instance of %s as constructor argument for %s is deprecated as of 3.2 and will be removed in 4.0. Pass an instance of %s instead.', ParserCacheInterface::class, self::class, CacheItemPoolInterface::class), E_USER_DEPRECATED);
+                $cache = new ParserCacheAdapter($cache);
+            } elseif (!$cache instanceof CacheItemPoolInterface) {
+                throw new \InvalidArgumentException(sprintf('Cache argument has to implements %s.', CacheItemPoolInterface::class));
+            }
+        }
+
+        $this->cache = $cache ?: new ArrayAdapter();
```

Comment:
I think we should keep it as is. Previously, some sort of cache would always have been used. If we made the cache optional now, it would be possible that the performance decreased for some use cases.